src/activity_handlers.cpp

```diff
@@ -586,7 +586,7 @@ void activity_handlers::longsalvage_finish( player_activity *act, player *p )
     }
     
     auto actor = dynamic_cast<salvage_actor*>( use_fun->get_actor_ptr() );
```

Comment:
I forgot one thing here: this line should not be executed if `use_fun` is `nullptr` (because then it will crash).
This doesn't happen in normal usage, but could happen if inventory was shifted.